### PR TITLE
Disable out of constraint dates in DateInputV2

### LINKED ISSUE #6353: Disable out of constraint dates in DateInput
**Describe the bug**
On selecting dates out of max & min constraint, DateInputV2 does not show any error to user.
Cannot differentiate between enabled and disabled dates

image
The dates marked are out of constraint but cannot be differentiated.

**To Reproduce**

1. Go to any page that has DateInput. 
   Ex: [https://care.coronasafe.in/facility/1210229b-711a-46e6-ab0a-207c4808e32e/assets/59bf0e46-0ea6-4a78-9d6f-b6c87395bb82](https://care.coronasafe.in/facility/1210229b-711a-46e6-ab0a-207c4808e32e/assets/59bf0e46-0ea6-4a78-9d6f-b6c87395bb82)
2. Scroll down to service history and click on Edit

**Expected behavior**
Dates out of constraint should be disabled and user could get an error message to not to select dates out of min and max

**Desktop (please complete the following information):**

- OS: Windows
- Browser: Chrome

### PR DESCRIPTION
**Pull request by suprabathk**
### <samp>🤖 Generated by Copilot at 91a9847</samp>

The pull request improves the `DateInputV2` component to support custom error messages and translations for date inputs, and to show valid and invalid dates more clearly. It also updates the English locale file with the new strings used by the component.

## Proposed Changes

- Fixes #6353 
- Added error notification when selecting out of constraint date.
- UI changes for disabled dates.
- Added translations.

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### <samp>🤖 Generated by Copilot at 91a9847</samp>

*  Add optional prop `outOfLimitsErrorMessage` to `DateInputV2` component to customize error message when date is out of range ([link](https://github.com/coronasafe/care_fe/pull/6379/files?diff=unified&w=0#diff-277f47613d077cedcaf477f502fdbe6f452f7da1410e81ac6741b61b255d8f7bR31))
*  Import `Notification` and `t` modules to display error notifications and translate strings in `DateInputV2` component ([link](https://github.com/coronasafe/care_fe/pull/6379/files?diff=unified&w=0#diff-277f47613d077cedcaf477f502fdbe6f452f7da1410e81ac6741b61b255d8f7bR17-R18))
*  Modify `setDateValue` function to check date constraints and show error notification if date is invalid ([link](https://github.com/coronasafe/care_fe/pull/6379/files?diff=unified&w=0#diff-277f47613d077cedcaf477f502fdbe6f452f7da1410e81ac6741b61b255d8f7bL108-R125))
*  Change `placeholder` props of `input` elements to use `t` function with keys `select_date` and `DD/MM/YYYY` for translation ([link](https://github.com/coronasafe/care_fe/pull/6379/files?diff=unified&w=0#diff-277f47613d077cedcaf477f502fdbe6f452f7da1410e81ac6741b61b255d8f7bL231-R240), [link](https://github.com/coronasafe/care_fe/pull/6379/files?diff=unified&w=0#diff-277f47613d077cedcaf477f502fdbe6f452f7da1410e81ac6741b61b255d8f7bL267-R276))
*  Change `className` prop of `div` element that renders date cells to conditionally apply styles based on date validity ([link](https://github.com/coronasafe/care_fe/pull/6379/files?diff=unified&w=0#diff-277f47613d077cedcaf477f502fdbe6f452f7da1410e81ac6741b61b255d8f7bL358-R374))
*  Add English translations for `select_date` and `DD/MM/YYYY` keys to `src/Locale/en/Common.json` file ([link](https://github.com/coronasafe/care_fe/pull/6379/files?diff=unified&w=0#diff-38a71477ce052dd5f81bf55434622f86d274a8dc6a8af1a03fb444588aec4296L155-R158))

### CODE REVIEW

**Comment by nihal467** [comment]
LGTM

File: src/Components/Common/DateInputV2.tsx [hunk 1]
```diff
@@ -355,11 +364,14 @@ const DateInputV2: React.FC<Props> = ({
...
-                                  : "hover:bg-gray-300",
-                                !isDateWithinConstraints(d) && "!text-gray-300"
+                                "flex h-full items-center justify-center rounded text-center text-sm leading-loose text-black transition duration-100 ease-in-out",
+                                isDateWithinConstraints(d)
+                                  ? value &&
+                                      (isSelectedDate(d)
+                                        ? "bg-primary-500 font-bold text-white"
+                                        : "hover:bg-gray-300") +
+                                        " cursor-pointer"
+                                  : "!cursor-not-allowed bg-gray-200 !text-gray-400"
```
Comment (khavinshankar) [inline src/Components/Common/DateInputV2.tsx]: @suprabathk simplify this logic, make it more readable

**Comment by nihal467** [comment]
@suprabathk what is the status on this PR, its been stale with no activity for a while now

**Comment by rithviknishad** [comment]
@Ashesh3 
image

**Comment by nihal467** [comment]
@Ashesh3 the issue is not solved

**Comment by Ashesh3** [comment]
> @Ashesh3 the issue is not solved

Should be fixed now.
image

**Comment by nihal467** [comment]
LGTM

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**